src/index.js

```diff
@@ -0,0 +1,168 @@
+		}
+
+		const { editEntityRecord } = useDispatch( 'core' );
+
+		const { selectedTerms, terms, taxonomy, postType, postId } = useSelect(
+			( select ) => {
+				const { getTaxonomy, getEntityRecords } = select( coreStore );
+
+				// Get the full alert level taxonomy object.
+				const alertLevelTaxonomy = getTaxonomy( slug );
+
+				// Get information about the current post.
+				const {
+					getEditedPostAttribute,
+					getCurrentPostType,
+					getCurrentPostId,
+				} = select( 'core/editor' );
+
+				return {
+					selectedTerms: alertLevelTaxonomy
+						? getEditedPostAttribute(
+								alertLevelTaxonomy.rest_base
+						  ).filter( ( val ) => {
+								return Number( val );
+						  } )
+						: [],
+					terms:
+						getEntityRecords( 'taxonomy', slug, {
+							per_page: -1,
+							orderby: 'name',
+							order: 'asc',
+							_fields: 'id,name',
+							context: 'view',
+						} ) || [],
+					taxonomy: alertLevelTaxonomy,
+					postType: getCurrentPostType(),
+					postId: getCurrentPostId(),
+				};
+			},
+			[]
+		);
+
+		// Parse available terms into a structure expected by the Select interface.
+		const termData = terms.map( ( term ) => {
+			return {
+				label: decodeEntities( term.name ),
+				value: term.id,
+			};
+		} );
+
+		const termsList = [
+			{
+				label: __( 'Select level' ),
+				value: 0,
+			},
+			...termData,
+		];
+
+		const [ meta, setMeta ] = useEntityProp( 'postType', postType, 'meta' );
+		const {
+			_hp_alert_display_through: displayThrough,
+			_hp_alert_has_expiration: hasExpiration,
+		} = meta;
+
+		let displayThroughValue;
+
+		if ( ! displayThrough ) {
+			displayThroughValue = new Date();
+		} else {
+			// Prep the display through value for JS Date compatibility. Microseconds!
+			displayThroughValue = new Date( displayThrough * 1000 );
+		}
+
+		// Update the post with the selected alert level.
+		const onChange = ( termID ) => {
+			const data = {};
+
+			if ( 0 === Number( termID ) ) {
+				setMeta( {
+					_hp_alert_has_expiration: false,
+					_hp_alert_display_through: '',
+				} );
+			}
+
+			data[ taxonomy.rest_base ] = [ parseInt( termID, 10 ) ];
+			editEntityRecord( 'postType', postType, postId, data );
+		};
+
+		return (
+			<>
+				<SelectControl
+					label={ __( 'Alert level' ) }
+					multiple={ false }
+					onChange={ onChange }
+					options={ termsList }
+					value={ selectedTerms }
+				/>
+				{ 0 < selectedTerms.length && (
+					<RadioControl
+						label={ __( 'Alert expires' ) }
+						selected={ hasExpiration ? 'yes' : 'no' }
+						options={ [
+							{
+								label: 'Yes',
+								value: 'yes',
+							},
+							{
+								label: 'No',
+								value: 'no',
+							},
+						] }
+						onChange={ ( value ) => {
+							if ( 'no' === value ) {
+								setMeta( {
+									_hp_alert_has_expiration: false,
+									_hp_alert_display_through: '',
+								} );
+							} else {
+								setMeta( {
+									_hp_alert_has_expiration: true,
```

Comment:
It's not quite clear to me if/how the `_hp_alert_has_expiration` meta is going to be leveraged outside of this interface, but that's probably just the lack of coffee speaking 😅.

Reply:
This was my (hacky) way of making sure the calendar interface is hidden and no meta is accidentally stored for the display through date if no expiration is necessary. It could probably be a block attribute instead 🤔

Reply:
Actually! I think this can be handled entirely through `hp_alert_display_through` because I'm storing a `0` value.

Reply:
Ah - yeah, a block attribute or maybe a `state`. Or, it's fine the way it is! 😀

Reply:
Took it out in b1137b7a1ca8c1b1b32abe9331ed832a7781fd84 and now rely on the `0` value of the expiration date instead. ✂️ 🎉